### DISCUSSION TITLE
MOCK_METHOD arguments limit of 15?

### DISCUSSION THREAD

**Discussion by PiotrNycz** [google/googletest#4602] [Community Help] [answered]
Has MOCK_METHOD macro hard limit on number of arguments to 15? 
If yes, where is that documented? 
If not documented - then maybe it should be. 
The compilation error is pretty hard to understand...

**Comment by PiotrNycz** [comment]
Looks it has:
[gmock/internal/gmock-pp.h](https://github.com/google/googletest/blob/main/googlemock/include/gmock/internal/gmock-pp.h)

```
// Requires: the number of arguments after expansion is at most 15.
#define GMOCK_PP_NARG(...) \
  GMOCK_PP_INTERNAL_16TH(  \
      (__VA_ARGS__, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0))
```

but just want confirmation this is desired behavior - and what about documenting it?

**Reply by bmaxdk** [reply, marked as answer]
Yes upto 15. [This article](https://www.apriorit.com/dev-blog/468-gmock-enchancements) mentions that the MOCK_METHOD macro traditionally has a hard limit of 10 arguments due to internal constraints, but an extension was created to allow up to 15 arguments.

I agree with you. It's better to documented with this limit in the official documentation to help to understand
